File: atcoder.jp/contests/arc105/tasks/arc105_a/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const sort = <T extends number | bigint>(array: T[], reverse = false) =>
  reverse
    ? array.slice().sort((a, b) => (a > b ? -1 : a < b ? 1 : 0))
    : array.slice().sort((a, b) => (a < b ? -1 : a > b ? 1 : 0));

const main = () => {
  const ABCD = input().split(" ").map(Number);

  const [A, B, C, D] = sort(ABCD);
  const ans = A + B + C === D || A + D === B + C;

  console.log(ans ? "Yes" : "No");
};

main();

File: atcoder.jp/contests/arc105/tasks/arc105_a/main.cpp
#include <bits/stdc++.h>
using namespace std;

template <typename T>
vector<T> sort(const vector<T> &vec, bool reverse = false) {
  vector<T> ret = {vec.begin(), vec.end()};
  if (reverse)
    sort(ret.begin(), ret.end(), greater());
  else
    sort(ret.begin(), ret.end());
  return ret;
}

vector<int> input_vector(int n) {
  vector<int> ret(n);
  for (auto &&i : ret) cin >> i;
  return ret;
}

int main() {
  auto ABCD = sort(input_vector(4));
  int A = ABCD[0];
  int B = ABCD[1];
  int C = ABCD[2];
  int D = ABCD[3];

  bool ans = A + B + C == D || A + D == B + C;
  cout << (ans ? "Yes" : "No") << endl;
}
